Library/Homebrew/formula.rb

```diff
@@ -476,6 +476,29 @@ def compiler_failures
     active_spec.compiler_failures
   end
 
+  # The best spec we can upgrade the {Formula} to so it can satisfy dependency.
+  def best_upgrade_spec_sym
```

Comment:
> Don't care about LinkedKegs and just the opt/foo link i.e. don't care about brew link status.

You mean for detecting the current spec? `Formulary#from_rack` should use `LinkedKegs` symlink if it's available and if it's not, then use optlink. But if they both exist, they are the same.